boot/bootutil/src/loader.c

```diff
@@ -234,11 +234,11 @@ boot_read_header_from_scratch(struct image_header *out_hdr)
 
     rc = flash_area_read(fap, 0, out_hdr, sizeof *out_hdr);
     if (rc != 0) {
-        flash_area_close(fap);
-        return BOOT_EFLASH;
+        rc = BOOT_EFLASH;
     }
 
-    return 0;
+    flash_area_close(fap);
```

Comment:
This is good, but we should probably squash it into the previous commit, so that we don't subtly break git bisect.